### PR TITLE
Fix PWA auth relay sync and release flow

### PR DESCRIPTION
**Pull request by silverbucket**
## Summary
- Fix standalone/PWA auth popup handling so login state can sync back into the app reliably.
- Add an auth relay page to support cross-window authentication handoff.
- Update remote storage/auth integration and add coverage for the sync flow.
- Switch CI and release workflows to `npm ci`, and commit release artifacts from committed source before tagging.
- Track `package-lock.json` and align release/deploy behavior with the committed build output.

## Testing
- `npm test`
- `npm run lint`
- Verified auth relay and popup handling changes via the new `src/lib/remotestorage.test.js` coverage.
- Not run: full end-to-end browser validation of the PWA auth flow in a production build.

### CODE REVIEW

File: src/lib/remotestorage.test.js [hunk 1]
```diff
@@ -248,3 +310,134 @@ describe("authorizeWithStandalonePopup", () => {
...
+            setTimeout: vi.fn(() => 2),
+            clearTimeout: vi.fn(),
+        };
+
+        const repo = createRemoteStorageRepository();
+
+        repo.connect("user@example.com");
+
+        expect(globalThis.window.open).toHaveBeenCalledTimes(1);
+        expect(globalThis.window.open).toHaveBeenCalledWith("", "_blank", "popup=yes,width=480,height=720");
```
[the remark this answers is not in the document]
Comment (silverbucket) [reply src/lib/remotestorage.test.js:581]: Adjusted the test to assert the popup URL/target plus feature substrings instead of the full literal feature string, so it follows the behavior without coupling to the exact constant formatting. Re-ran `npm run lint`, `npm test`, and `npm run build`.

File: src/lib/remotestorage.js [hunk 2]
```diff
@@ -208,11 +252,17 @@ export function createRemoteStorageRepository() {
         connect(userAddress, token) {
             // Re-enable caching in case it was reset by a previous disconnect
             remoteStorage.caching.enable(`/${APP_SCOPE}/`);
+            if (!token) {
+                reserveStandaloneAuthPopup();
+            } else {
+                releaseReservedAuthPopup({ close: true });
+            }
             remoteStorage.connect(userAddress, token);
```
[the remark this answers is not in the document]
Comment (silverbucket) [reply src/lib/remotestorage.js:261]: Fixed. `connect()` now normalizes `token` to `undefined`, uses the normalized value for popup reservation, and forwards the normalized token into `remoteStorage.connect()`. Re-ran `npm run lint`, `npm test`, and `npm run build`.

**Review by silverbucket — COMMENTED** [review]
## Review Notes

The core fix is sound — reserving the popup during `connect()` (user gesture context) and reusing it during `authorize()` is the canonical iOS Safari workaround.

### What's good
- Correct root-cause fix for iOS PWA popup blocking
- Graceful degradation with `location.replace()` / `.href` fallback
- Nice UX: "Preparing login…" placeholder in the blank popup
- Clean cleanup discipline — no leaked popups in error/disconnect paths
- Good test coverage for the reserve → connect → authorize → reuse flow
- `AUTH_POPUP_FEATURES` constant avoids string duplication

### Suggestions (non-blocking)

1. **Edge case: user closes reserved popup before `authorize()` fires.** If the user dismisses the "Preparing login…" popup, `releaseReservedAuthPopup()` returns a closed window, and `authorizeWithStandalonePopup` receives a dead popup. The fallback `env.open()` won't help either since it's not in a gesture context. The user gets a confusing error with no obvious recovery. Consider checking `popup.closed` before navigating.

2. **Missing test coverage for `switchTo()` and `disconnect()`.** Both call the new popup reservation/cleanup functions but aren't tested. Will add these.

**Comment by silverbucket** [comment]
Addressed the new review note.

1. The reserved-popup edge case is already handled by the current implementation: `releaseReservedAuthPopup()` returns `null` when the held popup is already closed, so `authorizeWithStandalonePopup()` never receives a dead window to navigate. In that case it falls back to the normal popup-open path and emits the existing auth error if the browser blocks it. I did not change behavior there.

2. Added the missing coverage for the new popup lifecycle around `switchTo()` and `disconnect()` in `src/lib/remotestorage.test.js`.

Verification re-run:
- `npm run lint`
- `npm test`
- `npm run build`

**Comment by silverbucket** [comment]
Followed up on the latest review note.

1. The standalone override was indeed missing part of the library's normal authorize setup. I restored the missing `redirectUri` / `clientId` derivation before building the popup auth URL, which is the actual bug in the `authorize({ authURL })` path.

2. On the reserved-popup edge case: the current `releaseReservedAuthPopup()` path already returns `null` for a closed held popup, so the popup reuse path does not try to navigate a dead window. I left that behavior unchanged.

3. Added the missing lifecycle coverage for `switchTo()` and `disconnect()`, and rewrote the auth-path tests to match the real library flow (`connect()` leading to `authorize({ authURL })`).

Verification re-run:
- `npm run lint`
- `npm test`
- `npm run build`
